docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,422 @@
+     }
+  });
+};
+
+const MyNewVisType = (Private) => {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    requestHandler: myRequestHandler
+  });
+}
+
+VisTypesRegistryProvider.register(MyNewVisType);
+-----------
+
+[[development-visualization-response-handlers]]
+=== Visualization Response Handlers
+Response handler will receive the data from request handler and an instance of Vis object. 
+Its job is to convert the data to a format visualization can use. By default 'default' request handler is used 
+which produces a table representation of the data. The data object will then be passed to visualization.
+
+[[development-default-response-handler]]
+==== default response handler
+Default response handler will convert pure elasticsearch response to tabular format.
+
+[[development-none-response-handler]]
+==== none response handler
+None response handler is an identity function, which will return the same data it receives.
+
+[[development-custom-response-handler]]
+==== custom response handler
+You can define your custom response handler by providing a function with the following definition:
+'function (vis, response) { ... }'.
+
+Function should return the transformed data object that visualization can consume.
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+const myResponseHandler = (vis, response) => {
+   // transform the response (based on vis object?)
+   if (vis.params.transformTable) return transformTable(response);
+   return response;
+};
+
+const MyNewVisType(Private) => {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    responseHandler: myResponseHandler
+  });
+}
+
+VisTypesRegistryProvider.register(MyNewVisType);
+-----------
+
+[[development-vis-object]]
+=== Vis object
+Vis object holds the visualization state and is the window into kibana:
+
+- *vis.params*: holds the visualization parameters
+- *vis.indexPattern*: selected index pattern object
+- *vis.getState()*: gets current visualization state 
+- *vis.updateState()*: updates current state with values from `vis.params`
+- *vis.resetState()*: resets `vis.params` to the values in the current state
+- *vis.forceReload()*: forces whole cycle (request handler gets called)
+- *vis.getUiState()*: gets UI state of visualization
+- *vis.uiStateVal(name, val)*: updates a property in UI state
+- *vis.isEditorMode()*: returns true if in editor mode
+- *vis.API.timeFilter*: allows you to access time picker
+- *vis.API.events.click*: default click handler
+- *vis.API.events.brush*: default brush handler
+
+Visualization gets all its parameters in `vis.params` (defaults merged with the current state). 
```

Comment:
The visualization gets all its parameters in `vis.params`, which are default values merged with the current state.